spec/controllers/followings_controller_spec.rb

```diff
@@ -24,6 +24,118 @@
       expect(assigns(:followings)).to eq(alice.followings)
     end
 
-    it 'redirects to the singin path if the user is not authenticated'
+    it 'orders the followings such that last followings shows first' do
+      jose = Fabricate(:user)
+      carlos = Fabricate(:user)
+      following1 = Fabricate(:following, user_id: alice.id, subject_id: jose.id)
+      following2 = Fabricate(:following, user_id: alice.id, subject_id: carlos.id)
+      get :index, user_id: alice.id
+      expect(assigns(:followings)).to match_array([following2, following1])
+    end
+
+    it_behaves_like 'require_sign_in' do
+      let(:user) {Fabricate(:user)}
+      let(:action) {get :index, user_id: alice.id} 
+    end
+  end
+
+  describe 'POST create' do
+    let(:alice) {Fabricate(:user)}
+    before{session[:user_id] = alice.id}
+
+    it 'redirects to the index action' do
+      joe  = Fabricate(:user)
+      post :create, subject_id: joe.id 
+      expect(response).to redirect_to(user_followings_path(alice))
+    end
+
+    it 'creates a following' do
+      joe  = Fabricate(:user)
+      post :create, subject_id: joe.id 
+      expect(Following.count).to eq(1)
+    end
+
+    it 'creates a following related with the current user' do
+      joe  = Fabricate(:user)
+      post :create, subject_id: joe.id 
+      expect(Following.first.user).to eq(alice)
+    end
+
+    it 'creates a following related with a subject' do
+      joe  = Fabricate(:user)
+      post :create, subject_id: joe.id 
+      expect(Following.first.subject).to eq(joe)
+    end
+
+    it 'sets a flash message if the following is succesfully created' do
+      joe  = Fabricate(:user)
+      post :create, subject_id: joe.id 
+      expect(flash[:notice]).to be_present
+    end
+
+    it 'does not follow the same subject twice' do
+      joe  = Fabricate(:user)
+      Fabricate(:following, user_id: alice.id, subject_id: joe.id)
+      post :create, subject_id: joe.id 
+      expect(Following.count).to eq(1)
+    end
+
+    it 'does not allow current user to follow himself' do
+      post :create, subject_id: alice.id 
+      expect(Following.count).to eq(0)
+    end
+
+    it 'sets a flash message if the following is succesfully created' do
+      joe  = Fabricate(:user)
+      Fabricate(:following, user_id: alice.id, subject_id: joe.id)
+      post :create, subject_id: joe.id 
+      expect(flash[:error]).to be_present
+    end
+
+    it_behaves_like 'require_sign_in' do
+      let(:user) {Fabricate(:user)}
+      let(:action) {post :create, user_id: alice.id, subject_id: user.id} 
+    end
+  end
+
+  describe 'DELETE destroy' do
+    let(:alice) {Fabricate(:user)}
+    before{session[:user_id] = alice.id}
+
+    it 'sets the following to be destroyed' do
```

Comment:
The next spec technically can't pass if this one fails....this one is a bit of overkill in my opinion. You should only test for instance variables when they will be used in a view.